contentcuration/contentcuration/frontend/shared/vuex/channel/actions.js

```diff
@@ -118,10 +120,48 @@ export function commitChannel(
         channelData.content_defaults = contentDefaults;
       }
     }
+
     return Channel.createModel(channelData).then(() => {
       context.commit('UPDATE_CHANNEL', { id, ...channelData });
       context.commit('SET_CHANNEL_NOT_NEW', id);
     });
+  } else {
+    // Create a new channel
+    const newChannelData = {};
+
+    if (name !== NOVALUE) {
+      newChannelData.name = name;
+    }
+    if (description !== NOVALUE) {
+      newChannelData.description = description;
+    }
+    if (language !== NOVALUE) {
+      newChannelData.language = language;
+    }
+    if (thumbnail !== NOVALUE) {
+      newChannelData.thumbnail = thumbnail;
+    }
+    if (thumbnail_url !== NOVALUE) {
+      newChannelData.thumbnail_url = thumbnail_url;
+    }
+    if (contentDefaults !== NOVALUE) {
+      newChannelData.content_defaults = contentDefaults;
+    }
```

Comment:
The above code is very similar to the `if` branch above in this function. I'm requesting deduplication of the code because the only real difference should be the presence of `id` which isn't significant enough to warrant the duplication. For example, it seems there are differences here in how it handles the thumbnail above, which leads to a regression that I can see when you upload a thumbnail for a new channel and adjust it's cropping.